engine/api/worker.go

```diff
@@ -114,6 +115,35 @@ func (api *API) postRegisterWorkerHandler() service.Handler {
 	}
 }
 
+func (api *API) getWorkerHandler() service.Handler {
+	return func(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+		vars := mux.Vars(r)
+		name := vars["name"]
+
+		withKey := FormBool(r, "withKey")
+
+		if !isCDN(ctx) {
```

Comment:
Need to check if this is enough. Maybe don't allow registering other service than hatchery for non admin.